src/main/kotlin/com/manimdsl/ASTExecutor.kt

```diff
@@ -14,35 +14,102 @@ data class StackValue(val initObject: MObject, val stack: Stack<Pair<Double, MOb
 
 object EmptyValue : ExecValue()
 
-class ASTExecutor(
-        private val program: ProgramNode,
-        private val symbolTableVisitor: SymbolTableVisitor,
-        private val fileLines: List<String>
-) {
+class VirtualMachine(private val program: ProgramNode, private val symbolTableVisitor: SymbolTableVisitor, private val statements: Map<Int, ASTNode>, private val fileLines: List<String>) {
 
     private val linearRepresentation = mutableListOf<ManimInstr>()
     private val variableNameGenerator = VariableNameGenerator(symbolTableVisitor)
-
-    private val codeBlockVariable: String
-    private val codeTextVariable: String
-    private val pointerVariable: String
+    private val codeBlockVariable: String = variableNameGenerator.generateNameFromPrefix("code_block")
+    private val codeTextVariable: String = variableNameGenerator.generateNameFromPrefix("code_text")
+    private val pointerVariable: String = variableNameGenerator.generateNameFromPrefix("pointer")
+    private val displayLine: MutableList<Int> = mutableListOf()
+    private val displayCode: MutableList<String> = mutableListOf()
 
     init {
-        pointerVariable = variableNameGenerator.generateNameFromPrefix("pointer")
-        codeBlockVariable = variableNameGenerator.generateNameFromPrefix("code_block")
-        codeTextVariable = variableNameGenerator.generateNameFromPrefix("code_text")
+        fileLines.indices.forEach {
+            val presentableLine = fileLines[it] == "}" || fileLines[it] == "{" || fileLines[it] == "" || (statements.containsKey(it + 1) && statements[it + 1] is CodeNode)
```

Comment:
Can this be abstracted into a function

Reply:
Just the condition to be a presentable line?

Reply:
You could just condense it to: `listOf("}", "{", "").contains(fileLines[it])` and then || the statement stuff?

Reply:
And then it's probably short enough to not be a function

Reply:
Also you probably don't need the contains key because you have the "is", so it could just be
`listOf("}", "{", "").contains(fileLines[it]) || statements[it + 1] is CodeNode` ?